controller/controller.go

```diff
@@ -39,11 +43,45 @@ func New(ctx *cli.Context) (*Controller, error) {
 		return nil, err
 	}
 
-	return NewWithDependencies(l, recorder), nil
+	var key crypto.PrivKey
+	identity := ctx.String("identity")
+	if !ctx.IsSet("identity") {
+		identity = ".dealbot.key"
+	}
+	if _, err := os.Stat(identity); os.IsNotExist(err) {
+		// make a new identity
+		pr, _, err := crypto.GenerateKeyPair(crypto.Ed25519, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		// save it.
+		b, err := crypto.MarshalPrivateKey(pr)
+		if err != nil {
+			return nil, err
+		}
+		if err := ioutil.WriteFile(identity, b, 0666); err != nil {
```

Comment:
this is a private key, so consider 0600 perhaps

Reply:
fixed in https://github.com/filecoin-project/dealbot/pull/57/commits/1e73b9e2ad2a0bb05c69ce267450609727cd24ef